File: leetcode/DP/leetcode_746_min_cost_climbing_stairs/solution.py
class Solution(object):
    def minCostClimbingStairs(self, cost):
        """
        :type cost: List[int]
        :rtype: int
        """
        dp = [0] * (len(cost)+1)

        dp[0] = 0
        dp[1] = 0

        for i in range(2, len(cost)+1):
            dp[i] = min((dp[i-1] + cost[i-1]), (dp[i-2] + cost[i-2] ))
        
        return dp[len(cost)]
    
solution = Solution()
cost = [1,100,1,1,1,100,1,1,100,1]
result = solution.minCostClimbingStairs(cost)
print(result)

File: leetcode/DP/leetcode_746_min_cost_climbing_stairs/Solution.java
package DP.leetcode_746_min_cost_climbing_stairs;

class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int[] dp = new int[cost.length + 1];
        dp[0] = 0;
        dp[1] = 0;

        for (int i = 2; i < cost.length + 1; i++) {
            dp[i] = Math.min(dp[i-2] + cost[i-2], dp[i-1] + cost[i-1]);
        }
        
        return dp[cost.length];
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // Example usage
        int[] cost = {1, 100, 1, 1, 1, 100, 1, 1, 100, 1};
        int result = solution.minCostClimbingStairs(cost);
        System.out.println(result);
    }
}
